src/ImageSharp/Formats/PixelTypeInfo.cs

```diff
@@ -4,50 +4,51 @@
 using System.Runtime.CompilerServices;
 using SixLabors.ImageSharp.PixelFormats;
 
-// TODO: Review this class as it's used to represent 2 different things.
+// TODO: Review this type as it's used to represent 2 different things.
 // 1.The encoded image pixel format.
 // 2. The pixel format of the decoded image.
 namespace SixLabors.ImageSharp.Formats;
 
 /// <summary>
 /// Contains information about the pixels that make up an images visual data.
 /// </summary>
-public class PixelTypeInfo
+/// <remarks>
+/// Initializes a new instance of the <see cref="PixelTypeInfo"/> struct.
+/// </remarks>
+/// <param name="bitsPerPixel">Color depth, in number of bits per pixel.</param>
+public readonly struct PixelTypeInfo(int bitsPerPixel)
 {
     /// <summary>
-    /// Initializes a new instance of the <see cref="PixelTypeInfo"/> class.
+    /// Gets color depth, in number of bits per pixel.
     /// </summary>
-    /// <param name="bitsPerPixel">Color depth, in number of bits per pixel.</param>
-    public PixelTypeInfo(int bitsPerPixel)
-        => this.BitsPerPixel = bitsPerPixel;
+    public int BitsPerPixel { get; init; } = bitsPerPixel;
 
     /// <summary>
-    /// Initializes a new instance of the <see cref="PixelTypeInfo"/> class.
+    /// Gets the count of the color components
     /// </summary>
-    /// <param name="bitsPerPixel">Color depth, in number of bits per pixel.</param>
-    /// <param name="alpha">The pixel alpha transparency behavior.</param>
-    public PixelTypeInfo(int bitsPerPixel, PixelAlphaRepresentation alpha)
-    {
-        this.BitsPerPixel = bitsPerPixel;
-        this.AlphaRepresentation = alpha;
-    }
+    public byte ComponentCount { get; init; }
```

Comment:
We should consider changing this to `int`.
```suggestion
    public int ComponentCount { get; init; }
```

Unless there is a reason to compress data for optimization (~the type is expected to be instantiated very frequently), it is generally recommended to prefer `int` over other integral types in .NET APIs. For example, `IPEndPoint.Port` is an `int`, while technically it could be an `ushort`. The reason is that this removes the need for conversion which is both expensive and complicated.

On the other hand, I noticed that we use non-`int` integrals in other descriptor types, eg. `PngMetadata`. I wish I noticed this earlier.

Reply:
I’ve often wondered if there was a rule. For v4 we can definitely fix this.

Reply:
All good. I've fixed the property for this PR, we can look at others.